### ISSUE TITLE
Fatal error: Missing column when fetching a nil belongsTo association

### ISSUE THREAD

**Issue by myyra** [groue/GRDB.swift#999]
### What did you do?

Tried to `fetchOne` a `belongsTo` association, where the column with the foreign key is `NULL`. `Self` and the `belongsTo` destination have the same backing table, but different structs.

### What did you expect to happen?

The `fetchOne` would return `nil`, like doing a `fetchOne(db, key: nil)`. Also calling `fetchAll` on the destination's corresponding `hasMany` works as expected, returning an empty array. Fetching with a non-`nil` key using the same association also works as expected, returning the referenced row.

### What happened instead?

Crash with `GRDB/SQLRelation.swift:945: Fatal error: Missing column: projectId`

### Environment

**GRDB flavor(s):** None
**GRDB version:** 5.8.0
**Installation method:** SPM
**Xcode version:** both 12.5.1 (12E507) and 13.0 beta (13A5155e)
**Swift version:** both 5.3 and 5.5
**Platform(s) running GRDB:** iOS
**macOS version running Xcode:** 11.4 (20F71)

### Demo Project

<details>
<summary>A self contained test case demonstrating the issue:</summary>

``` swift
import XCTest
import GRDB

enum ItemType: String, Codable {
    case task
    case project
}

struct Task: Codable, Equatable,TableRecord, FetchableRecord, PersistableRecord {

    static var databaseTableName = "item"

    let id: UUID
    var title: String
    var projectId: UUID?
    let type: ItemType

    static let project = belongsTo(Project.self, key: "project")
    var project: QueryInterfaceRequest<Project> {
        request(for: Self.project)
    }
}

struct Project: Codable, Equatable,TableRecord, FetchableRecord, PersistableRecord {

    static var databaseTableName = "item"

    let id: UUID
    let title: String
    let type: ItemType

    static let tasks = hasMany(Task.self, key: "task")
    var tasks: QueryInterfaceRequest<Task> {
        request(for: Self.tasks)
    }
}

class GRDBBugTests: XCTestCase {

    func makeDB() throws -> DatabaseQueue {
        let db = DatabaseQueue()

        var migrator = DatabaseMigrator()

        migrator.registerMigration("item") { db in
            try db.create(table: "item", body: { t in
                t.column("id", .blob).primaryKey()
                t.column("title", .text).notNull()
                t.column("projectId", .blob).references("item", onDelete: .cascade)
                t.column("type", .text).notNull()
            })
        }

        try migrator.migrate(db)

        return db
    }

    let task = Task(
        id: UUID(uuidString: "f943b43f-cd0a-4a90-b6dd-4f5f308e8db3")!,
        title: "Test Task",
        projectId: UUID(uuidString: "3f8b4368-090e-4ac6-b93f-3aefacaaa194")!,
        type: .task
    )

    let project = Project(
        id: UUID(uuidString: "3f8b4368-090e-4ac6-b93f-3aefacaaa194")!,
        title: "Test Project",
        type: .project
    )

    func testFetchWithKey() throws {
        let db = try makeDB()

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let fetchedProject = try db.read { db in
            try Project.fetchOne(db, key: task.projectId)
        }

        XCTAssertEqual(project, fetchedProject)
    }

    func testFetchAssociation() throws {
        let db = try makeDB()

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let fetchedProject = try db.read { db in
            try task.project.fetchOne(db)
        }

        XCTAssertEqual(project, fetchedProject)
    }

    func testFetchWithNilKey() throws {
        let db = try makeDB()

        var task = task
        task.projectId = nil

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let fetchedProject = try db.read { db in
            try Project.fetchOne(db, key: task.projectId)
        }

        XCTAssertEqual(nil, fetchedProject)
    }

    // Crashes
    func testFetchNilAssociation() throws {
        let db = try makeDB()

        var task = task
        task.projectId = nil

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let fetchedProject = try db.read { db in
            try task.project.fetchOne(db)
        }

        XCTAssertEqual(nil, fetchedProject)
    }

    func testFetchAssociationWithRequest() throws {

        struct TaskWithProject: Equatable, FetchableRecord, Codable {
            let task: Task
            let project: Project?
        }

        let db = try makeDB()

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let request = Task
            .filter(Column("type") == "task")
            .including(optional: Task.project)

        let taskWithProject = try db.read { db in
            try TaskWithProject.fetchOne(db, request)
        }

        XCTAssertEqual(taskWithProject!.project, project)
    }

    func testFetchNilAssociationWithRequest() throws {

        struct TaskWithProject: Equatable, FetchableRecord, Codable {
            let task: Task
            let project: Project?
        }

        let db = try makeDB()

        var task = task
        task.projectId = nil

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let request = Task
            .filter(Column("type") == "task")
            .including(optional: Task.project)

        let taskWithProject = try db.read { db in
            try TaskWithProject.fetchOne(db, request)
        }

        XCTAssertEqual(taskWithProject!.project, nil)
    }

    func testFetchTasks() throws {
        let db = try makeDB()

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let tasks = try db.read { db in
            try project.tasks.fetchAll(db)
        }

        XCTAssertEqual([task], tasks)
    }

    func testFetchNilTasks() throws {
        let db = try makeDB()

        var task = task
        task.projectId = nil

        try db.write { db in
            try project.save(db)
            try task.save(db)
        }

        let tasks = try db.read { db in
            try project.tasks.fetchAll(db)
        }

        XCTAssertEqual([], tasks)
    }
}
```

</details>

**Comment by groue** [comment]
Hello @myyra, thank you very much for the bug report and the reproducing case 👍 

The fix will ship shortly :-)

**Comment by groue** [comment]
I have difficulties setting up tests on Travis.com, and this prevents me from swiftly shipping the fix with the next version of the library. :-/

Meanwhile, you can find the fix in the development branch.

**Comment by myyra** [comment]
That's fine, thanks for the quick fix!

**Comment by groue** [comment]
Add thank you for sponsoring the project! ❤️

**Comment by groue** [comment]
The fix has shipped in [v5.9.0](https://github.com/groue/GRDB.swift/tree/v5.9.0).
